### ISSUE TITLE
Wrong Titleinformation to Sonos / Last.fm 

### ISSUE THREAD

**Issue by hosh0815** [music-assistant/support#3383]
### What version of Music Assistant has the issue?

2.3.4

### What version of the Home Assistant Integration have you got installed?

2024.12.5 (build in)

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

When I stream (doesn´t matter if local file share or Apple Music), the Titlefinormation track is scrobbeln and shown on Sonos WITH the Artist. So all played stuff is on the Last.fm side for track inforamtion "ARTIST - TRACKtitle". On my  Mac I installed the "menu bar", where you can see the Information also scrobbeld to last.fm

### How to reproduce

Streaming Music from MA to sonos connected with Last.fm account 

### Music Providers

Apple Music, local file share

### Player Providers

sonos

### Full log output

[music-assistant_lastfm.log](https://github.com/user-attachments/files/18273022/music-assistant_lastfm.log)

### Additional information

<img width="245" alt="SCR-20241230-ipst" src="https://github.com/user-attachments/assets/49c54045-15b5-4870-9e3f-e3324e3bf5e6" />
<img width="861" alt="SCR-20241230-ipuw" src="https://github.com/user-attachments/assets/ea3a26fa-7343-4c87-b339-666081931fd1" />

### What version of Home Assistant Core are your running

2024.12.5

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Alternative

**Comment by hosh0815** [comment]
here is a other example: correct in MA, additional artist in title in "Sonos" and Lastfm (this is how it is scrobbeld to Last.fm)
<img width="1236" alt="SCR-20241230-iyev" src="https://github.com/user-attachments/assets/2d9f1195-53e1-48af-928f-399aa3c92193" />

**Comment by hosh0815** [comment]
and now this: 

<img width="829" alt="SCR-20241230-jcvi" src="https://github.com/user-attachments/assets/00e278dc-30be-4a80-b2b3-8a8fb756a9b4" />

So, for some tracks it works, for some not... all from the same provider and same tagging (streaming from apple music should be tagged correct) 

I don´t expect anything, perhaps some help or tips for searching

in the log file is this shown:
AttributeError: 'NoneType' object has no attribute 'get'
2024-12-30 09:04:27.037 WARNING (MainThread) [music_assistant.webserver] Disconnected: Received non-Text message.
2024-12-30 09:32:28.821 INFO (MainThread) [music_assistant] Loaded metadata provider The Audio DB
2024-12-30 09:36:31.947 INFO (MainThread) [music_assistant.apple_music] No top tracks found for artist Sia

**Comment by hosh0815** [comment]
<img width="811" alt="grafik" src="https://github.com/user-attachments/assets/11311da2-03fa-4acf-bb64-a0df974b262e" />

**Comment by OzGav** [comment]
MA doesn’t support scrobble so I am not sure how you are doing this. Since MA doesn’t support scrobbling this doesn’t seem to be a MA issue?

**Comment by hosh0815** [comment]
that´s what I ask for: somewhere in the process (MA -> SONOS -> LastFM scrobbling) must be the time, when the IDTag getting the additional Artist in the Songtitle... So when it only would happening on playing from local files = tagged mostly correct, but there could be the wrong ID Tag.. but, from Apple music there must be the correct IDTag for scrobbling. But there is no difference between both providers.

Now I tried to deactivate the Database for metadata, but with no changes. Is there a possibility, to change the IDTags?

**Comment by OzGav** [comment]
Sorry I am not following you at all. Your report title is that the incorrect information is sent to Sonos and last.fm

MA does not send any information to last.fm so this is confusing. Looking back at your post are you saying that in the SONOS app the track name is actually artist-track? I don’t know what that interface is which you show in the very first screenshot?

**Comment by hosh0815** [comment]
yes, in MA the ID Tag Title and Artist is shown correct, so there must be the delivery to SONOS / Last.FM with wrong or additional informations... any idea where this could be ?

**Comment by OzGav** [comment]
No because MA streams to the player. I can only assume the Sonos app gets information from the player. I’ll need to ask the devs for more information.

**Comment by hosh0815** [comment]
ok, would be nice. after an update from sonos devices I have now this behaviour: 
the entitys of the Sonos speaker now showing the Title with the artist, in MA there´s not. So I would say, that the metadata whitch are transfered from MA to Sonos are filled with to much data or a checkbox is marked wrongly... But I don´t know where to find.... I tried to disable "The Audio DB", but no change...

<img width="1152" alt="grafik" src="https://github.com/user-attachments/assets/b392e2b4-aa17-4ebc-a8a5-6d3169d30019" />

**Comment by OzGav** [comment]
I strongly suggest you don’t disable things in MA if you don’t know what it does. AudioDB is used to add missing metadata to existing tracks, it has nothing to do with how information is sent to the players.

**Comment by hosh0815** [comment]
of course I made backups and just disabled for one track... but, no result, so I stopped trying. 

This is what I´m searching for: where is the point, the MA is sending informations to the player and can I have some changes in the way it does or what is transmitted? thank you: happy new year

**Comment by hosh0815** [comment]
Hello again; I´ve just found out, that the behaviour starts with installing a beta, after the "normal" didn´t work well (because of the LAN-connected SONOS... but, after installing the built-in MA, it doesn´t change... so, what do you suggest... complete new installing or is there a checkbox to change something
![grafik](https://github.com/user-attachments/assets/f86adfa7-8ad0-4009-a08b-cda481d283a4)

**Comment by hosh0815** [comment]
so now I completly wiped my installation and started from one... after I started playing from Apple Music/FileShare, the ID Tag become wrong / additional the Artist in the Tracktitle... so can you do something with this information?

**Comment by OzGav** [comment]
This will get looked at further when workload allows.

**Comment by hosh0815** [comment]
hey: I downgraded my hole Sonos System to S1... now it works all fine again... MA transfers the correct Title Tag to Sonos: Sonos correct to Last.fm :) 

so: case closed for me... dunno if this is a real "problem"

**Comment by OzGav** [comment]
We can’t control Sonos firmware so if this has fixed it for you then all good.

**Comment by marcelveldt** [comment]
This is not fixed, this is an actual issue on our end we need to fix. You just had to be patient, there was absolutely no reason to go back to S1 for a simple title being wrong

**Comment by hosh0815** [comment]
Back on S2. The behavior is back again. Don't care about, I saw that there is work on a scrobbling tool from music assistant directly. I appreciate that very much. :)

**Comment by hosh0815** [comment]
Hey, with the Latest Beta Updateit works fine again 

Thx

**Comment by hosh0815** [comment]
beta 7 resolved... or somethin different... it works again...
